### DISCUSSION TITLE
How to work with subscriptions, real time data and notifications

### DISCUSSION THREAD

**Discussion by reymon359** [blitz-js/blitz#2743] [Q&A] [answered]
Hi there 👋🏽 

I am building an app with Blitz.js and until now everything was going fine, actually, the development has been fast and easy, and I loved it. However, soon or later I will have to work with real-time data, subscriptions, and notifications for users. AFAIK, Blitz.js has no support for this yet. I was thinking about polling from time to time, but I just don't like that approach when there are better ones.

Therefore, I am considering moving to another stack now that my project is still in the early days. I want to keep using TypeScript and Next.js on the frontend and PostgreSQL as the DB. As for the backend/API itself, I don't know what to use. I plan to have a look at Hasura, which works with PostgreSQL and seems to have the real-time data under control. Plus,  it has already a lot of built-in solutions by their community.

I am also considering making my own GraphQL-API  where I can handle the subscriptions, but this will mean more work on my own. However, maybe it's better than creating a dependency on Hasura. Or just use Next.js API as the backend (probably this would be the easiest to manage) 🤔

Has someone had a similar issue with Blitz working with real-time data and subscriptions? 

Any tip, comment, or suggestion is more than welcomed! 😄

**Comment by flybayer** [comment, marked as answer]
Hey sorry for delay in responding.

If I was in your situation, I would keep using Blitz but add in Hasura or something to use just for the cases where you need real-time. That way you get the productivity and power of Blitz while also having realtime. And once Blitz has better realtime story, you can switch from Hasura to that.

**Reply by reymon359** [reply]
Thanks for the answer @flybayer and the suggestion.

And don't worry about the delay, I appreciate that you take the time to answer as much as you can while improving the community. 😄
